lib/dao.js

```diff
@@ -468,6 +468,18 @@ DataAccessObject.updateOrCreate = DataAccessObject.upsert = function upsert(data
   var Model = this;
   var connector = Model.getConnector();
 
+  if (Array.isArray(data)) {
+    var isBulkCreate = data.every(function(item){
+      var id = getIdValue(self, item);
+      return id === undefined || id === null;
+    });
+    if (isBulkCreate) {
+      return self.create(data, options, cb);
+    } else {
```

Comment:
proably don't need the else since you return in the if

```
if (isBulkCreate) {
  return self.create(data, options, cb);
}
return cb(new Error('...'));
```